File: src/main/java/com/deepexi/interview_system/service/InterviewService.java
package com.deepexi.interview_system.service;


import com.deepexi.interview_system.dto.PageResult;
import com.deepexi.interview_system.dto.RequestParam;
import com.deepexi.interview_system.model.Company;
import com.deepexi.interview_system.model.Interviewee;

import java.util.List;

public interface InterviewService {
    /**
     * 查询所有
     * @return
     */
    List<Interviewee> findAll();

    /**
     * 根据条件分页查询
     * @param pageNum
     * @param size
     * @param requestParam
     * @return
     */
    PageResult findPage(int pageNum , int size, RequestParam requestParam);

    /**
     * 根据id查询公司
     */
    List<Company> findCompanyById(Integer applicant_id);

    /**
     * 插入interview对象
     * @param interviewee
     */
    void insertInterview(Interviewee interviewee);

    /**
     * 更新interview对象
     * @param interviewee
     */
    void updateInterview(Interviewee interviewee);

    /**
     * 根据id查询所有信息
     * @param id
     * @return
     */
    Interviewee findInterviewById(int id);

    /**
     * 根据id删除信息
     * @param aids
     */
    void delete(int aids);

    /**
     * 批量删除
     * @param ids
     */
    void deleteIds(int[] ids);

    /**
     * 面试官下拉数据
     * @return
     */
    List<String> findInterviewer();

    /**
     * 面试职位下拉数据
     * @return
     */
    List<String> findInterviewpost();

}

File: src/main/resources/static/js/service/interviewService.js
app.service("interviewService",function ($http) {
    //查询所有
    this.findAll = function () {
        return $http.get('/interview/findAll');
    }

    //分页查询
    this.findPage = function (page,size) {
        return $http.get("/interview/findPage/"+page+"/"+size);
    }

    //搜索
    this.search = function (page,size,requestParam) {
        return $http.post("/interview/search/"+page+"/"+size,requestParam);
    }

    //新增
    this.add = function (entity) {
        return $http.post("/interview/insert",entity);
    }

    //根据id查询
    this.findOne = function (id) {
        return $http.get("/interview/findOne/"+id);
    }

    // 修改
    this.update = function (entity) {
        return $http.post("/interview/update",entity);
    }

    this.deleteSelectIds = function (ids) {
        return $http.delete("/interview/delete/"+ids);
    }

    this.findInterviewer = function () {
        return $http.get("/interview/findInterviewer");
    }

    this.findInterviewpost = function () {
        return $http.get("/interview/findInterviewpost");
    }
})